openage/convert/terrain_convert.pyx

```diff
@@ -8,66 +8,89 @@ Transforms the dimetric tiles converted from SLP files to cartesian format.
 
 import math
 import argparse
-from PIL import Image
 from .slp import SLP
 from .texture import Texture
+from PIL import Image
+import numpy as np
 
 class Terrain_convert:
-
+    
+    # Method to simply put the tile on the big merge img. Something is wrong on this method because the result is transparent.
+    # TO FIX.
+    @staticmethod
+    def replace(overlayer, overlayed, overlayer_x, overlayer_y, overlayed_x, overlayed_y, slp_width, slp_height):
+        to_ret = overlayed
+        xr = overlayer_x
+        xd = overlayed_x
+        yr = overlayer_y
+        yd = overlayed_y
+        while xd < overlayed.shape[0]:
+            while yd < overlayed.shape[1]:
+                # if the overlayer is not transparent, then we can push it on the overlayed
+                # without the if, we would overwrite non-transparent pixels with transparent ones.
+                if xr < overlayer.shape[0] and yr < overlayer.shape[1] and overlayer[xr,yr][3]>0:
+                    to_ret[xd,yd] = overlayer[xr,yr]
+                yd+=1
+                yr+=1
```

Comment:
`yd` is never reset, so the inner loop is only executed once.

Also, what is the purpose of the `xr < overlayer.shape[0]` and `yr < overlayer.shape[1]` conditions?

Reply:
Wow I feel stupid now haha, of course I have to reset them! 
Because the size of the merge_img is a bit smaller than the original SLP terrain texture, I had to make sure I wasn't pasting any pixel out of the merge_img. Basically, these conditions are there to avoid out of bound indexes.

Reply:
Ah! However, I still wonder what the inner and outer loop accomplish exactly. Wouldn't the current logic operate on the whole image in the worst case (`xd, yd = 0`) instead of just one tile?

Reply:
Ah! However, I still wonder what the inner and outer loop accomplish exactly. Wouldn't the current logic operate on the whole image in the worst case (`xd, yd = 0`) instead of just one tile?

Reply:
That's what I corrected, thanks to your advices. It's now merging the tiles but the coordinates are wrong. I have to correct them.